### PR TITLE
New serverless pattern - apigw-openapi-lambda-dynamodb-xray-terraform

### PR DESCRIPTION
**Pull request by nikhilswagle**
*Issue #, if available:*

*Description of changes:*
This pattern demonstrates how to deploy an API Gateway REST API with Lambda backend integration that is defined in an OpenAPI specification format using Terraform. The pattern also shows how backend Lambda is integrated with Lambda powertools for logging and more importantly tracing the requests end to end from API Gateway to DynamoDB data store.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by singledigit — CHANGES_REQUESTED** [review]
There are several problems with this pattern.

1. There is not an `example-pattern.json` file in the root.
2. We are having users run a deploy script. This is not a preferred method. We have several options. Either use the serverless.tf modules or have TF run the script.

Please refactor and add an example-pattern file.

**Comment by nikhilswagle** [comment]
Requested changes have been made.

Thanks,
Damodar

File: apigw-openapi-lambda-dynamodb-xray-terraform/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+  "title": "Deploying API Gateway REST API using OpenAPI spec with Lambda Powertools",
```
Comment (singledigit) [inline apigw-openapi-lambda-dynamodb-xray-terraform/example-pattern.json:2]:
```suggestion
  "title": "Amazon API Gateway to AWS Lambda using OpenAPI",
```

File: apigw-openapi-lambda-dynamodb-xray-terraform/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    "text": [
+      "terraform -chdir=infra destroy"
+    ]
+  },
+  "authors": [
+    {
+      "name": "Damodar Shenvi Wagle",
+      "image": "https://drive.google.com/file/d/1o-NvTy-Ock12bKOLoMiwyolqizNNMhv8/view?usp=sharing",
+      "bio": "Damodar Shenvi Wagle is a Senior Cloud Application Architect at AWS Professional Services.",
+      "linkedin": "www.linkedin.com/in/damodar-shenvi-wagle-72375548"
```
Comment (singledigit) [inline apigw-openapi-lambda-dynamodb-xray-terraform/example-pattern.json:66]:
```suggestion
      "linkedin": "damodar-shenvi-wagle-72375548"
```

**Review by singledigit — APPROVED** [review]

**Pull request merged**